src/itemDeclaredChoice/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { Component, Fragment } from 'react'
+
+import Item from '_utils/item'
+import { color } from '_utils/branding'
+import { TextDisplayType } from 'text'
+import Loader from 'loader'
+import CircleIcon from 'icon/circleIcon'
+import CheckIcon from 'icon/checkIcon'
+
+import style from './style'
+
+export enum ItemDeclaredChoiceStatus {
+  DEFAULT = 'default',
+  LOADING = 'loading',
+}
+
+export interface ItemDeclaredChoiceProps {
+  readonly label: string
+  readonly name: string
+  readonly value: string | number
+  readonly data?: string
+  readonly className?: Classcat.Class
+  readonly labelTitle?: string
+  readonly dataInfo?: string
+  readonly checked?: boolean
+  readonly disabled?: boolean
+  readonly onChange?: (obj: OnChangeParameters) => void
+  readonly status?: ItemDeclaredChoiceStatus
+  readonly key?: string | number
+}
+
+class ItemDeclaredChoice extends Component<ItemDeclaredChoiceProps> {
+  static defaultProps: Partial<ItemDeclaredChoiceProps> = {
+    onChange() {},
+    checked: false,
+  }
+
+  static STATUS = ItemDeclaredChoiceStatus
+
+  onChange = () => {
+    const { name, value } = this.props
+    this.props.onChange({ name, value })
+  }
+
+  render() {
+    const {
+      label,
+      name,
+      value,
+      data,
+      className,
+      labelTitle,
+      dataInfo,
+      checked,
+      disabled,
+      status,
+    } = this.props
+    const isLoading = status === ItemDeclaredChoiceStatus.LOADING
+
+    let radioIcon = <CircleIcon iconColor={color.primary} thin />
+    if (checked) {
+      radioIcon = <CheckIcon iconColor={color.white} backgroundColor={color.primary} thin />
+    }
+    if (isLoading) {
+      radioIcon = <Loader size={24} inline />
+    }
```

Comment:
The loading status logic is similar, but ToggleButton has a totally different html output (being a button, and not an input radio for accessibility). But maybe I can factorize this logic somehow... I'm just scared of over-factorizing, while the code duplication is actually minor.

Reply:
Yes I missed that part, then maybe this logic should be held in a new internal component on its own. We can do this in another PR, what do you think?

Reply:
I've extracted the little checkbox "icon" with its status into an internal component for consistency across components. This will make sure we have the same design for these :)